metashare/api/jobs.py

```diff
@@ -55,6 +55,19 @@ def report_errors_on_provision(scratch_org):
         raise
 
 
+@contextlib.contextmanager
+def report_errors_on_check_changes(scratch_org):
+    try:
+        yield
+    except Exception as e:
+        async_to_sync(report_scratch_org_error)(
+            scratch_org, str(e), "SCRATCH_ORG_FETCH_CHANGES_FAILED"
+        )
+        tb = traceback.format_exc()
+        logger.error(tb)
+        raise
```

Comment:
Is this a separate function for testability, or what? Since it's used in exactly one place I'd find it more readable if it were just implemented inline there.

Reply:
I think this should also set `has_changes: false` -- since it represents the fact that we can't reliably tell whether the org currently has changes or not. Either it'll need to fire a `SCRATCH_ORG_UPDATE` event, or make that change before the `SCRATCH_ORG_FETCH_CHANGES_FAILED` payload is serialized.

Reply:
It's separate at this point because I think entering multiple contexts with an `ExitStack` is more legible than entwining multiple layers of setup and cleanup in a single `try-except`.